universal-login-react/src/core/services/WalletStorageService.ts

```diff
@@ -1,34 +1,65 @@
 import {StorageEntry, StorageService} from './StorageService';
-import {ApplicationWallet} from '@universal-login/commons';
-import {asObject, asString, Sanitizer} from '@restless/sanitizers';
-import {WalletStorage} from '@universal-login/sdk';
+import {ApplicationWallet, asExactly, CounterfactualWallet} from '@universal-login/commons';
+import {asAnyOf, asObject, asString, cast} from '@restless/sanitizers';
+import {WalletStorage, SerializedWalletState} from '@universal-login/sdk';
+
+const STORAGE_KEY = 'wallet';
```

Comment:
Should we use more sophisticated storage key, e.g. `wallet.jarvis.network`?

Reply:
This is a good idea, but Jarvis already uses the `wallet` key for storage. We might want to consider some more sophisticated migration mechanisms if we decide to change the key or add versioning

Reply:
https://github.com/EthWorks/original-works/tree/master/original-works-mobile/src/services/migrations

Reply:
Exactly, but with YAGNI, I would start with a very simple migration mechanism (just one method) and evolve it into a full-blown migration mechanism if needed. 
It can be done in future PRs.